src/gen_js_api.ml

```diff
@@ -444,55 +456,91 @@ let parse_attr ~global_attrs ?ty (s, loc, auto) (k, v) =
   | exception Not_found -> None
   | _, f -> Some (f ())
 
-let parse_valdecl ~global_attrs ~in_sig vd =
+let parse_require_attribute ~require_ctx ~default attributes =
+  let require =
+    match get_attribute "js.require" attributes with
+    | None -> None
+    | Some (k, v) ->
+        if require_ctx then error k.loc (Not_supported_here "nested requires")
+        else
+          match v with
+          | PStr [] -> Some default
+          | _ -> Some (id_of_expr (expr_of_payload k.loc v))
+  in
+  require, require_ctx || require <> None
+
```

Comment:
I am not sure it's a good idea that this function returns a pair, since the second projection can be simply obtained from the first component and the require_ctx argument. My guess is that it would simplify the code a bit to move the corresponding logic at call sites (when needed).